main/src/server.rs

```diff
@@ -328,24 +361,26 @@ impl ServiceBuilder {
             .copied()
             .expect("Config grpc_listen_addr cannot be empty.");
 
-        GrpcService::new(
+        Some(GrpcService::new(
             self.runtime.clone(),
             kv,
             coord,
             addr,
             None,
             self.metrics_register.clone(),
             self.span_context_extractor.clone(),
-        )
+        ))
     }
 
-    fn create_vector_grpc(
+    fn create_vector_grpc_if_enabled(
         &self,
         coord: CoordinatorRef,
         dbms: DBMSRef,
-        port: u16,
-    ) -> VectorGrpcService {
-        let default_vector_grpc_addr = build_default_address(port);
+    ) -> Option<VectorGrpcService> {
+        let default_vector_grpc_addr = match self.config.cluster.vector_listen_port {
```

Comment:
Is it better to put this judgment outside

Reply:
Is this pattern better?

```rust
fn create_vector_grpc(
    &self,
    coord: CoordinatorRef,
    dbms: DBMSRef,
    port: u64,
) -> VectorGrpcService {
   ...
}

if let Some(port) = config.cluster.vector_listen_port {
    let vector_service = create_vector_grpc(...);
    server.add_service(Box::new(vector_service));
}
```